src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala

```diff
@@ -1408,4 +1456,224 @@ trait SilverTransforms extends SparkSessionWrapper {
+            ),
+          lit(0) // don't allow null returns
+        )
+      )
+      .withColumn(
+        "target_num_clusters", // need to check this logic and rename column
+        coalesce(
+          when(!'isRunning || 'isRunning.isNull, lit(null).cast("long"))
+            .when('state === "CREATING",
+              coalesce('min_num_clusters, 'current_num_clusters))
+            .otherwise(coalesce('max_num_clusters, 'current_num_clusters)),
+          lit(0) // don't allow null returns
+        )
+      )
+      .select(
+        'organization_id, 'warehouse_id, 'isRunning,
+        'timestamp, 'state, 'current_num_clusters, 'target_num_clusters
+      )
+      .withColumn("unixTimeMS_state_start", 'timestamp)
+      .withColumn("unixTimeMS_state_end", coalesce( // if state end open, use pipelineSnapTime, will be merged when state end is received
+        lead('timestamp, 1).over(stateUnboundW) - lit(1), // subtract 1 millis
+        lit(untilTime.asUnixTimeMilli)
+      ))
+
+    // Start changes from here
+    // Get the warehouseID that has been Permenantly_Deleted
+
+    val removedWarehouseID = warehouseBaseDF
+      .filter('actionName.isin("deleteEndpoint"))
+      .select('warehouse_id,'timestamp.alias("deletion_timestamp")).distinct()
+
+    val warehouseEventsBaselineForRemovedCluster = warehouseEventsBaseline.join(removedWarehouseID,Seq("warehouse_id"))
+
+    val window = Window.partitionBy('organization_id, 'warehouse_id).orderBy('timestamp.desc)
+    val stateBeforeRemoval = warehouseEventsBaselineForRemovedCluster
+      .withColumn("rnk",rank().over(window))
+      .withColumn("rn", row_number().over(window))
+      .withColumn("unixTimeMS_state_end",when('state.isin("STOPPING","TERMINATING_IMPUTED"),'unixTimeMS_state_end).otherwise('deletion_timestamp))
+      .filter('rnk === 1 && 'rn === 1).drop("rnk", "rn")
+
+    val stateDuringRemoval = stateBeforeRemoval
+      .withColumn("timestamp",when('state.isin("STOPPING","TERMINATING","TERMINATING_IMPUTED"),'unixTimeMS_state_end+1).otherwise(col("deletion_timestamp")+1))
+      .withColumn("isRunning",lit(false))
+      .withColumn("unixTimeMS_state_start",('timestamp))
+      .withColumn("unixTimeMS_state_end",('timestamp))
+      .withColumn("state",lit("PERMENANT_DELETE"))
+      .withColumn("current_num_clusters",lit(0))
+      .withColumn("target_num_clusters",lit(0))
+      .drop("deletion_timestamp")
+
+    val columns: Array[String] = warehouseEventsBaseline.columns
+    val stateDuringRemovalFinal = stateBeforeRemoval.drop("deletion_timestamp")
+      .unionByName(stateDuringRemoval, allowMissingColumns = true)
+      .select(columns.map(col): _*)
+
+    val warehouseEventsBaselineFinal = warehouseEventsBaseline.join(stateDuringRemovalFinal,Seq("warehouse_id","timestamp"),"anti")
+      .select(columns.map(col): _*)
+      .unionByName(stateDuringRemovalFinal, allowMissingColumns = true)
+
+    warehouseEventsBaselineFinal
+      .withColumn("counter_reset",
```

Comment:
This column name is difficult to understand what it's referring to. Even if this is a table we don't expect customers query directly, it shouldn't be so cryptic.

Reply:
done